src/components/SearchFormComponent/index.jsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { useState } from 'react'
+import JSONDATA from './data.json'
+
+export const SearchForm = () => {
+    const [searchTerm, setSearchTerm] = useState('')
+    return (
+    <>
+        <input type="text" placeholder="Search any Movie..." onChange={event => {setSearchTerm(event.target.value)}} />
+        <button type="submit">Search</button>
+        {JSONDATA.filter((val)=> {
```

Comment:
Couple of points here:
1. Consider always creating a separate named function for method callbacks/handlers.
E.g. `JSONDATA.filter(filterHandler).map(mapHandler);`
2. Consider avoiding putting logic in JSX directly, use function calls or variable instead.
The difference between function call and variable would be the difference in placement relative to the component body itself: function calls for things from the outside and variable for things inside the body.
E.g. `const renderSearchTerms = JSONDATA.filter(filterSearchTerms).map(renderSearchTerm);`
`<> ... {renderSearchTerms}... </>`

Reply:
Separate functions have been created for the filter and map handlers. 
Variable has also been created for render search terms.